Src/BootCamp.Chapter/Player.cs

```diff
@@ -73,34 +77,116 @@ public Item[] GetItems(string name)
+            }
+            if (weightIfEquipped <= EquipCapacity)
+            {
+                _equipment.Chest = chest;
+            }
         }
 
-        public void Equip(Shoulderpiece head, bool isLeft)
+        public void Equip(Shoulderpiece shoulder, bool isLeft)
         {
-
+            var weightIfEquipped = WeightIfAdded(shoulder);
+            if (isLeft)
+            {
+                if (_equipment.LeftShoulder != null)
+                {
+                    weightIfEquipped -= _equipment.LeftShoulder.Weight;
+                }
+                if (weightIfEquipped <= EquipCapacity)
+                {
+                    _equipment.LeftShoulder = shoulder;
+                }
+            }
+            else
+            {
+                if (_equipment.RightShoulder != null)
+                {
+                    weightIfEquipped -= _equipment.RightShoulder.Weight;
+                }
+                if (weightIfEquipped <= EquipCapacity)
+                {
+                    _equipment.RightShoulder = shoulder;
+                }
+            }
         }
 
-        public void Equip(Legspiece head)
+        public void Equip(Legspiece legs)
         {
-
+            var weightIfEquipped = WeightIfAdded(legs);
+            if (_equipment.Legs != null)
+            {
+                weightIfEquipped -= _equipment.Legs.Weight;
+            }
+            if (weightIfEquipped <= EquipCapacity)
+            {
+                _equipment.Legs = legs;
+            }
         }
 
-        public void Equip(Armpiece head, bool isLeft)
+        public void Equip(Armpiece arms, bool isLeft)
         {
-
+            var weightIfEquipped = WeightIfAdded(arms);
+            if (isLeft)
+            {
+                if (_equipment.LeftArm != null)
+                {
+                    weightIfEquipped -= _equipment.LeftArm.Weight;
+                }
+                if (weightIfEquipped <= EquipCapacity)
+                {
+                    _equipment.LeftArm = arms;
+                }
+            }
+            else
+            {
+                if (_equipment.RightArm != null)
+                {
+                    weightIfEquipped -= _equipment.RightArm.Weight;
+                }
+                if (weightIfEquipped <= EquipCapacity)
+                {
+                    _equipment.RightArm = arms;
+                }
+            }
         }
 
-        public void Equip(Gloves head)
+        public void Equip(Gloves gloves)
         {
+            var weightIfEquipped = WeightIfAdded(gloves);
+            if (_equipment.Gloves != null)
+            {
+                weightIfEquipped -= _equipment.Gloves.Weight;
+            }
+            if (weightIfEquipped <= EquipCapacity)
+            {
+                _equipment.Gloves = gloves;
+            }
+        }
 
+        float WeightIfAdded(Item item)
+        {
+            return _equipment.GetTotalWeight() + item.Weight;
```

Comment:
If you're equipping a null item (which pretty much means unequipping!) then this would throw a `NullReferenceExecption`